kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyDeployment.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.kubernetes.operator;
+
+import java.util.Map;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.ContainerBuilder;
+import io.fabric8.kubernetes.api.model.PodTemplateSpec;
+import io.fabric8.kubernetes.api.model.PodTemplateSpecBuilder;
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.api.model.apps.DeploymentBuilder;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+
+/**
+ * The Kube {@code Deployment} for the proxy
+ */
+@KubernetesDependent
+public class ProxyDeployment
+        extends CRUDKubernetesDependentResource<Deployment, KafkaProxy> {
+
+    public static final String CONFIG_VOLUME = "config-volume";
+    public static final String CONFIG_PATH_IN_CONTAINER = "/opt/kroxylicious/config/config.yaml";
+    public static final Map<String, String> APP_KROXY = Map.of("app", "kroxylicious");
+
+    public ProxyDeployment() {
+        super(Deployment.class);
+    }
+
+    static String deploymentName(KafkaProxy primary) {
+        return primary.getMetadata().getName();
+    }
+
+    @Override
+    protected Deployment desired(KafkaProxy primary,
+                                 Context<KafkaProxy> context) {
+        return new DeploymentBuilder()
+                .editOrNewMetadata()
+                .withName(deploymentName(primary))
+                .withNamespace(primary.getMetadata().getNamespace())
+                .withLabels(deploymentLabels())
+                .endMetadata()
+                .editOrNewSpec()
+                .withReplicas(1)
+                .editOrNewSelector()
+                .withMatchLabels(deploymentSelector())
+                .endSelector()
+                .withTemplate(podTemplate(primary))
+                .endSpec()
+                .build();
```

Comment:
The uniform indentation imposed by the formatter is going to really harm the readability benefits of using the builder APIs. I want to use indentation in denote nesting, but I'm getting overruled.

Reply:
I didn't think the formatter was being so zealous it was overriding custom line breaks/extra indentation. Though I guess it doesn't see the nesting as custom...

Reply:
Yeah, I might have been a bit overeager to blame the formatter here...

Reply:
Nope, it turns out that it really is the formatter :-(